subsys/esb/esb.c

```diff
@@ -1103,14 +1182,10 @@ void esb_disable(void)
 	memset(rx_pipe_info, 0, sizeof(rx_pipe_info));
 	memset(pids, 0, sizeof(pids));
 
-	/*  Disable the interrupts used by ESB */
+	/* Disable the interrupts used by ESB */
 	irq_disable(RADIO_IRQn);
-	irq_disable(ESB_SYS_TIMER_IRQn);
+	irq_disable(ESB_TIMER_IRQ);
 	irq_disable(ESB_EVT_IRQ);
-
```

Comment:
`nrf_radio_event_clear(NRF_RADIO, NRF_RADIO_EVENT_DISABLED);` should be added here to make sure there are no lingering events.

Reply:
What about adding it in the `esb_stop_rx` function which triggers disable task?

Reply:
Yes, I think that is sufficient